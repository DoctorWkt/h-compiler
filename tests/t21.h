void main()
{
  int x;
  for (x=-6; x != 7; x++) {
    if (x)		// Test that this works
      printn(x);
  }
  putchar('\n');

  for (x=-6; x != 7; x++) {
    if (x)		// Test that this works
      printn(x);
    else
      printn(45);
  }
  putchar('\n');

  x=-6;
  while (x) {		// Test that this works
    printn(x); x++;
  }
  putchar('\n');

  x= -6;
  do {
    printn(x); x++;
  } while (x);		// Test that this works
  putchar('\n');

  for (x=-6; x; x++) {	// Test that this works
    printn(x);
  }
  

}
