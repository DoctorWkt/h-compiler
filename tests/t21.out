777772
777773
777774
777775
777776
777777
000001
000002
000003
000004
000005
000006

777772
777773
777774
777775
777776
777777
000055
000001
000002
000003
000004
000005
000006

777772
777773
777774
777775
777776
777777

777772
777773
777774
777775
777776
777777

777772
777773
777774
777775
777776
777777
